File: gamepack/Decompiled/LocSound.java
public final class LocSound extends Node {
   static byte[][] jw;
   static NodeDeque af = new NodeDeque();
   RawPcmStream stream2;
   RawPcmStream stream1;
   LocType obj;
   int aa;
   int ab;
   int ac;
   int soundEffectId;
   int an;
   int aq;
   int au;
   int aw;
   int ax;
   int ay;
   int[] soundEffectIds;

   LocSound() {
   }

   void set() {
      int var2 = 1671764667 * this.soundEffectId;
      LocType var3 = this.obj.multiLoc();
      if (null != var3) {
         this.soundEffectId = var3.ambientSoundId * -853803453;
         this.aq = -981504640 * var3.int4;
         this.aa = -1756454329 * var3.int5;
         this.ay = -1723341761 * var3.int6;
         this.soundEffectIds = var3.bc;
      } else {
         this.soundEffectId = -24779379;
         this.aq = 0;
         this.aa = 0;
         this.ay = 0;
         this.soundEffectIds = null;
      }

      if (var2 != 1671764667 * this.soundEffectId && this.stream1 != null) {
         fe.pcmStreamMixer.removeSubStream(this.stream1);
         this.stream1 = null;
      }

   }
}

File: updater-runestar/src/main/java/org/runestar/client/updater/mapper/identifiers/classes/LocSound.kt
package org.runestar.client.updater.mapper.identifiers.classes

import org.objectweb.asm.Opcodes.GETFIELD
import org.objectweb.asm.Type.INT_TYPE
import org.runestar.client.updater.mapper.abstractclasses.IdentityMapper
import org.runestar.client.updater.mapper.abstractclasses.OrderMapper
import org.runestar.client.updater.mapper.abstractclasses.UniqueMapper
import org.runestar.client.updater.mapper.annotations.DependsOn
import org.runestar.client.updater.mapper.annotations.MethodParameters
import org.runestar.client.updater.mapper.predicateutilities.and
import org.runestar.client.updater.mapper.predicateutilities.predicateOf
import org.runestar.client.updater.mapper.predicateutilities.type
import org.runestar.client.updater.mapper.wrappers.Class2
import org.runestar.client.updater.mapper.wrappers.Field2
import org.runestar.client.updater.mapper.wrappers.Instruction2
import org.runestar.client.updater.mapper.wrappers.Method2

//@DependsOn(Node::class, LocType::class)
//class LocSound : IdentityMapper.Class() {
//
//    override val predicate = predicateOf<Class2> { it.superType == type<Node>() }
//            .and { it.instanceFields.any { it.type == type<LocType>() } }
//
//    @DependsOn(LocType::class)
//    class obj : IdentityMapper.InstanceField() {
//        override val predicate = predicateOf<Field2> { it.type == type<LocType>() }
//    }
//
//    @MethodParameters()
//    class set : IdentityMapper.InstanceMethod() {
//        override val predicate = predicateOf<Method2> { true }
//    }
//
//    @DependsOn(set::class)
//    class soundEffectId : OrderMapper.InMethod.Field(set::class, 0) {
//        override val predicate = predicateOf<Instruction2> { it.opcode == GETFIELD && it.fieldType == INT_TYPE }
//    }
//
//    class soundEffectIds : IdentityMapper.InstanceField() {
//        override val predicate = predicateOf<Field2> { it.type == IntArray::class.type }
//    }
//
//    @DependsOn(set::class, RawPcmStream::class)
//    class stream1 : UniqueMapper.InMethod.Field(set::class) {
//        override val predicate = predicateOf<Instruction2> { it.opcode == GETFIELD && it.fieldType == type<RawPcmStream>() }
//    }
//
//    @DependsOn(stream1::class, RawPcmStream::class)
//    class stream2 : IdentityMapper.InstanceField() {
//        override val predicate = predicateOf<Field2> { it.type == type<RawPcmStream>() && it != field<stream1>() }
//    }
//}